app/controllers/OverviewController.scala

```diff
@@ -233,36 +242,58 @@ class OverviewController @Inject()(
         .addingToSession((Constants.SRN, srn.value))
     }
 
-  def onSelectStart(srn: Srn, taxYear: String, version: String, reportType: String): Action[AnyContent] =
-    identify.andThen(allowAccess(srn)).andThen(getData).andThen(createData).async { implicit request =>
-      reportType match {
-        case PsrReportType.Sipp.name =>
-          val sippUrl = s"${config.urls.sippBaseUrl}/${srn.value}${config.urls.sippStartJourney}"
-          Future.successful(
-            Redirect(sippUrl)
-              .addingToSession(Constants.TAX_YEAR -> taxYear)
-              .addingToSession(Constants.VERSION -> version)
-          )
-        case _ =>
-          val yearFrom = LocalDate.parse(taxYear)
-          val yearTo = yearFrom.plusYears(1).minusDays(1)
-          val dateRange = DateRange(yearFrom, yearTo)
-          for {
-            userAnswers <- Future.fromTry(request.userAnswers.set(WhichTaxYearPage(srn), dateRange))
-            _ <- saveService.save(userAnswers.copy(id = UNCHANGED_SESSION_PREFIX + userAnswers.id))
-            _ <- saveService.save(userAnswers)
-          } yield {
-            Redirect(controllers.routes.WhatYouWillNeedController.onPageLoad(srn, "", taxYear, version))
-          }
+  def onSelectStart(
+    srn: Srn,
+    taxYear: String,
+    version: String,
+    reportType: String,
+    lastSubmittedPsrFbInPreviousYears: Option[String]
+  ): Action[AnyContent] =
+    identify
+      .andThen(allowAccess(srn))
+      .andThen(createData)
+      .andThen(
+        prePopulatedData(
+          Option.when(reportType == PsrReportType.Standard.name)(lastSubmittedPsrFbInPreviousYears).flatten
+        )
+      )
+      .async { implicit request =>
+        reportType match {
+          case PsrReportType.Sipp.name =>
+            val sippUrl = s"${config.urls.sippBaseUrl}/${srn.value}${config.urls.sippStartJourney}"
+            Future.successful(
+              Redirect(sippUrl)
+                .addingToSession(Constants.TAX_YEAR -> taxYear)
+                .addingToSession(Constants.VERSION -> version)
+            )
+          case _ =>
+            val yearFrom = LocalDate.parse(taxYear)
+            val yearTo = yearFrom.plusYears(1).minusDays(1)
+            val dateRange = DateRange(yearFrom, yearTo)
+            for {
+              userAnswers <- Future.fromTry(request.userAnswers.set(WhichTaxYearPage(srn), dateRange))
+              _ <- saveService.save(userAnswers.copy(id = UNCHANGED_SESSION_PREFIX + userAnswers.id))
+              _ <- saveService.save(userAnswers)
+            } yield {
+              if (lastSubmittedPsrFbInPreviousYears.isDefined) {
+                Redirect(controllers.routes.WhatYouWillNeedController.onPageLoad(srn, "", taxYear, version))
+                  .addingToSession(Constants.PREPOPULATION_FLAG -> String.valueOf(true))
```

Comment:
@cyeniceri do you think it is safe to store this in the session, as this value will be used quite further down the line? Just thinking about the issues we seen in PB that can be due to the restart of the application (or any other reasons that clear out the session).
Should we store it in mongo FE cache instead? or in request?